src/core/primitives.rs

```diff
@@ -30,6 +30,39 @@ fn xor_in_place<const LENGTH: usize>(a: &mut [u8; LENGTH], b: &[u8; LENGTH]) {
     }
 }
 
+/// Provides authenticity check for a given user key.
```

Comment:
You do not check for integrity inside this function.
```suggestion
/// Computes the signature of the given user key.
```